src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -130,6 +134,11 @@ int BATT_SMBUS::task_spawn(int argc, char *argv[])
 		}
 	}
 
+	if (myoptind >= argc) {
```

Comment:
This is not required as you don't access `argv` anymore.

Reply:
I think its still good practice to leave this check

Reply:
No, because you will unnecessarily return if there is no more argument given and that's not what we want here.

Reply:
@bkueng : I used the same schema as in previus commits. for example:
https://github.com/PX4/Firmware/blob/9782aecc73b45ae72ce01e507b1535c43f3780b6/src/drivers/barometer/lps22hb/lps22hb_main.cpp#L273

do you see why not to use it here too?

Reply:
Because BATT_SMBUS is using a different structure (the ModuleBase class), and `BATT_SMBUS::task_spawn` is only called for the `start` command. This is the reason why you don't need to check for it in here.

Reply:
ok, change done. I wasn't aware to this behavior.